### PR TITLE
Fix names when using `MultiBlock.transform`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

`transform` for `MultiBlock` uses integer indexing with `enumerate` to update the blocks. It turns out that indexing this way does not preserve the block name, e.g.:
``` python
import pyvista as pv
poly = pv.PolyData()
m = pv.MultiBlock(dict(data=poly)) 

# Block name is 'data'
m.keys() # ['data'] 

# Block name is modified
m[0] = poly
m.keys() # ['Block-00']
```

And so the filter unexpectedly mutates the block names as a result. This PR fixes this by indexing by name instead so that output names (keys) match the input names.

### CODE REVIEW

File: tests/core/test_composite.py [hunk 1]
```diff
@@ -450,23 +450,28 @@ def test_transform_filter(ant, sphere, airplane, tetbeam, inplace):
     nested = multi_from_datasets(airplane, tetbeam)
     multi.append(nested)
     multi.append(None)
+    for i in range(len(multi)):
+        multi.set_block_name(i, str(i))
```
Comment (tkoyama010) [inline tests/core/test_composite.py:454]:
```suggestion
    for i in enuerate(multi):
        multi.set_block_name(i, str(i))
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — DISMISSED** [review]
LGTM

**Review by tkoyama010 — APPROVED** [review]

File: tests/core/test_composite.py [hunk 2]
```diff
@@ -450,23 +450,28 @@ def test_transform_filter(ant, sphere, airplane, tetbeam, inplace):
...
     # Do test
     output = multi.transform(
         transform, inplace=inplace, transform_all_input_vectors=False, progress_bar=False
     )
     bounds_after = np.array(output.bounds)
     n_blocks_after = output.n_blocks
+    keys_after = output.keys()
 
     assert (output is multi) == inplace
     assert [(block_in is block_out) == inplace for block_in, block_out in zip(multi, output)]
```
Comment (user27182) [inline tests/core/test_composite.py:471]: Not directly related to this PR, but this looks like a truthy-object (always asserts True). Missing `all`
```suggestion
    assert all((block_in is block_out) == inplace for block_in, block_out in zip(multi, output))
```

Comment (tkoyama010) [reply tests/core/test_composite.py:471]: Good catch. It seems that we have more. Could you open new PR for this?
```bash
$ grep "assert \[" ./tests/**/*.py
./tests/core/test_composite.py:    assert [(block_in is block_out) == inplace for block_in, block_out in zip(multi, output)]
./tests/examples/test_dataset_loader.py:    assert [os.path.isabs(file) for file in path]
./tests/examples/test_dataset_loader.py:    assert [os.path.isabs(file) for file in path_loadable]
./tests/examples/test_dataset_loader.py:    assert [os.path.isfile(file) for file in path_download]
./tests/examples/test_dataset_loader.py:    assert [
./tests/examples/test_dataset_loader.py:    assert [
```

Comment (tkoyama010) [reply tests/core/test_composite.py:471]: I fixed it in #6588. I added you as co-author.

**Pull request merged**